### PR TITLE
Build on FreeBSD12 only

### PR DESCRIPTION
**Pull request by hasufell**
https://gitlab.haskell.org/maerwald/haskell-language-server/-/jobs/871886

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2420"></a>

### CODE REVIEW

**Review by jneira — APPROVED** [review]
LGTM, many thanks for fixing the gitlab builds

**Comment by hasufell** [comment]
Windows 8.8.4 fails: https://gitlab.haskell.org/maerwald/haskell-language-server/-/jobs/872984

```
ghc.exe: addLibrarySearchPath: C:\GitLabRunner\builds\0\872984\toolchain\ghcup\msys64\ming[2 of 2] Compiling Paths_haskell_language_server ( C:\GitLabRunner\builds\0\872984\dist-newstyle\build\x86_64-windows\ghc-8.8.4\haskell-language-server-1.5.1.0\x\haskell-language-server-wrapper\opt\build\haskell-language-server-wrapper\autogen\Paths_haskell_language_server.hs, C:\GitLabRunner\builds\0\872984\dist-newstyle\build\x86_64-windows\ghc-8.8.4\haskell-language-server-1.5.1.0\x\haskell-language-server-wrapper\opt\build\haskell-language-server-wrapper\haskell-language-server-wrapper-tmp\Paths_haskell_language_server.o )
Assembler messages:
Fatal error: can't create C:\\GitLabRunner\\builds\\0\\872984\\dist-newstyle\\build\\x86_64-windows\\ghc-8.8.4\\haskell-language-server-1.5.1.0\\x\\haskell-language-server-wrapper\\opt\\build\\haskell-language-server-wrapper\\haskell-language-server-wrapper-tmp\\Paths_haskell_language_server.o: No such file or directory
`gcc.exe' failed in phase `Assembler'. (Exit code: 1)
```

**Comment by hasufell** [comment]
https://gitlab.haskell.org/maerwald/haskell-language-server/-/pipelines/44701 looks good and should be finished soon

**Comment by jneira** [comment]
> Windows 8.8.4 fails: https://gitlab.haskell.org/maerwald/haskell-language-server/-/jobs/872984
> 
> ```
> ghc.exe: addLibrarySearchPath: C:\GitLabRunner\builds\0\872984\toolchain\ghcup\msys64\ming[2 of 2] Compiling Paths_haskell_language_server ( C:\GitLabRunner\builds\0\872984\dist-newstyle\build\x86_64-windows\ghc-8.8.4\haskell-language-server-1.5.1.0\x\haskell-language-server-wrapper\opt\build\haskell-language-server-wrapper\autogen\Paths_haskell_language_server.hs, C:\GitLabRunner\builds\0\872984\dist-newstyle\build\x86_64-windows\ghc-8.8.4\haskell-language-server-1.5.1.0\x\haskell-language-server-wrapper\opt\build\haskell-language-server-wrapper\haskell-language-server-wrapper-tmp\Paths_haskell_language_server.o )
> Assembler messages:
> Fatal error: can't create C:\\GitLabRunner\\builds\\0\\872984\\dist-newstyle\\build\\x86_64-windows\\ghc-8.8.4\\haskell-language-server-1.5.1.0\\x\\haskell-language-server-wrapper\\opt\\build\\haskell-language-server-wrapper\\haskell-language-server-wrapper-tmp\\Paths_haskell_language_server.o: No such file or directory
> `gcc.exe' failed in phase `Assembler'. (Exit code: 1)
> ```

that is due to the long path generated, we had to workaround in gh with this trick:

https://github.com/haskell/haskell-language-server/blob/1a74937ca51569fda63afeb4564deb94f5037c37/.github/workflows/test.yml#L133-L140

**Comment by hasufell** [comment]
> that is due to the long path generated, we had to workaround in gh with this trick:

I already fixed that

**Review by jneira — APPROVED** [review]
many thanks for fix the builds and make easy the distribution of artifacts

**Comment by hasufell** [comment]
postponing merge because I added commits that try to tar up the executables better... will have to see how it goes: https://gitlab.haskell.org/maerwald/haskell-language-server/-/pipelines/44715

**Comment by hasufell** [comment]
should be done

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -32,35 +36,67 @@ workflow:
       when: always
     - when: never
 
-.build:
-  script:
-    - bash .gitlab/ci.sh
+.artifacts:
```
Comment (hasufell) [inline .gitlab-ci.yml:39]: lolo @merijn

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -71,7 +71,7 @@ build-x86_64-linux-alpine:
 build-x86_64-freebsd:
   extends: .build
   tags:
-    - x86_64-freebsd13
+    - x86_64-freebsd12
```
Comment (hasufell) [inline .gitlab-ci.yml:150]: ya

**Comment by hasufell** [comment]
???

**Comment by jneira** [comment]
the branch has to be in sync with master to be merged

**Pull request merged**